src/Microsoft.Identity.Client/Platforms/Android/AndroidPlatformProxy.cs

```diff
@@ -40,38 +41,39 @@ namespace Microsoft.Identity.Client.Platforms.Android
     /// Platform / OS specific logic.  No library (ADAL / MSAL) specific code should go in here. 
     /// </summary>
     [global::Android.Runtime.Preserve(AllMembers = true)]
-    internal class AndroidPlatformProxy : IPlatformProxy
+    internal class AndroidPlatformProxy : AbstractPlatformProxy
     {
         internal const string AndroidDefaultRedirectUriTemplate = "msal{0}://auth";
 
-        private readonly Lazy<IPlatformLogger> _platformLogger = new Lazy<IPlatformLogger>(() => new AndroidPlatformLogger());
-        private IWebUIFactory _overloadWebUiFactory;
+        public AndroidPlatformProxy(ICoreLogger logger) : base(logger)
+        {
+        }
 
         /// <summary>
         /// Get the user logged in 
         /// </summary>
         /// <returns>The username or throws</returns>
-        public Task<string> GetUserPrincipalNameAsync()
+        public override Task<string> GetUserPrincipalNameAsync()
         {
             return Task.FromResult(string.Empty);
 
         }
-        public Task<bool> IsUserLocalAsync(RequestContext requestContext)
+        public override Task<bool> IsUserLocalAsync(RequestContext requestContext)
         {
             return Task.FromResult(false);
         }
 
-        public bool IsDomainJoined()
+        public override bool IsDomainJoined()
         {
             return false;
         }
 
-        public string GetEnvironmentVariable(string variable)
+        public override string GetEnvironmentVariable(string variable)
         {
             return null;
         }
 
-        public string GetProcessorArchitecture()
+        protected override string InternalGetProcessorArchitecture()
```

Comment:
>Internal [](start = 34, length = 8)

I don't understand the renaming to internal. probably we should talk in person on this

Reply:
Happy to talk.  These are the abstract methods that will get called to populate the lazy.  They have to have a different name than the public GetProcessorArchitecture (which resolves the lazy) so as to not have a naming conflict.  Open to better naming suggestions as well.

---
In reply to: [247619611](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/784#discussion_r247619611) [](ancestors = 247619611)